File: Work/ServerAnwendungen/EasyExportGeneralTask/EasyExportGeneralTask/EasyResult.cs
using System;
using System.Collections.Generic;
using System.Data;

namespace EasyExportGeneralTask
{
    public class EasyResult
    {
        /// <summary>
        /// Trefferzeilen-Zähler
        /// </summary>
        public int hitCounter { get; set; }

        /// <summary>
        /// Liste der Treffer
        /// </summary>
        public DataTable hitList { get; set; }

        public EasyResult()
        {
            clear();
        }

        /// <summary>
        /// Setzt die Result Tabelle zurück
        /// </summary>
        public void clear()
        {
            hitList = new DataTable {TableName = "HitList"};

            // Grundspalten ergänzen
            hitList.Columns.Add("DOC_Location", typeof(String));
            hitList.Columns.Add("DOC_Archive", typeof(String));
            hitList.Columns.Add("DOC_ID", typeof(String));
            hitList.Columns.Add("DOC_VERSION", typeof(String));
            hitList.Columns.Add("Bilder", typeof(String));
            hitList.Columns.Add("Filepath", typeof(String));
            hitList.Columns.Add("File", typeof(String));
            hitList.Columns.Add("FileLength", typeof(Int32));
        }

        /// <summary>
        /// Fügt eine Reihe von Spalten zur Ergebnis-Tabelle hinzu falls diese noch nicht existieren
        /// </summary>
        /// <param name="lst">Liste der Spaltennamen</param>
        public void AddColumnsToResultTable(List<string> lst)
        {
            foreach (string item in lst)
            {
                if (!hitList.Columns.Contains(item))
                {
                    hitList.Columns.Add(item);
                }
            }
        }

        /// <summary>
        /// Erzeugt aus einem EasyArchiv Header-String eine Liste der vorhandenen Spalten
        /// </summary>
        /// <param name="header">unformatierter Header-String aus EasyArchiv-Abfrage</param>
        /// <returns>Liste von Spaltennamen in Easyarchiv-Reihenfolge</returns>
        public List<string> CreateHeaderFieldlist(string header)
        {
            string[] headerSplit = header.Replace("^", "").ToUpper().Split(',');

            return new List<string>(headerSplit);
        }

        /// <summary>
        /// Fügt der Ergebnistabelle eine neue Row hinzu und füllt die angegebenen Spalten mit Werten aus dem Row-String
        /// </summary>
        /// <param name="row">Easy-Archiv DatenRow als String</param>
        /// <param name="columnlist">Spaltenschema der Row als Liste</param>
        /// <param name="strLocFound">EasyArchiv Location</param>
        /// <param name="strArcFound">EasyArchiv Archivname</param>
        /// <param name="doc_id">EasyArchiv DocumentID</param>
        /// <param name="doc_version">EasyArchiv Documentversion</param>
        public void addRowToResultTable(string row, List<string> columnlist, 
            string strLocFound, string strArcFound, string doc_id, string doc_version)
        {
            string[] SplittedRow = row.Replace("^,^", ";").Replace("^", "").Split(';');

            DataRow nRow = hitList.NewRow();

            nRow["DOC_Location"] = strLocFound;
            nRow["DOC_Archive"] = strArcFound;
            nRow["DOC_ID"] = doc_id;
            nRow["DOC_VERSION"] = doc_version;
            nRow["Bilder"] = null;

            for (int i = 0; i < SplittedRow.GetLength(0); i++)
            {
                nRow[columnlist[i]] = SplittedRow[i];
            }

            hitList.Rows.Add(nRow);
            hitList.AcceptChanges();
        }
    }
}

File: dev/work/Tools/EasyAccess40/EasyAccess40/EasyResult.vb
Imports System.Collections

Public Class EasyResult

    Private numOfHits As Long
    Private numOfFields As Long
    Private hitLst As Data.DataTable
    Private hitTableHeader As List(Of EasyResultField)

    Public Sub New()
        hitLst = New Data.DataTable()
        hitTableHeader = New List(Of EasyResultField)
    End Sub

    Public Property hits() As String
        Get
            Return numOfHits
        End Get
        Set(ByVal Value As String)
            numOfHits = Value
        End Set
    End Property

    Public Property fields() As String
        Get
            Return numOfFields
        End Get
        Set(ByVal Value As String)
            numOfFields = Value
        End Set
    End Property

    Public Property hitList() As Data.DataTable
        Get
            Return hitLst
        End Get
        Set(ByVal Value As Data.DataTable)
            hitLst = Value
        End Set
    End Property

    Public Property hitTblHeader() As List(Of EasyResultField)
        Get
            Return hitTableHeader
        End Get
        Set(ByVal Value As List(Of EasyResultField))
            hitTableHeader = Value
        End Set
    End Property

    Public Function getHitTable() As Data.DataTable
        Return hitLst
    End Function

    Public Sub clear()
        hitLst = New Data.DataTable()
    End Sub

    Public Sub addHitTableHeader(ByVal header As String)

        header = header.Replace("^", "").ToUpper

        'Position der Ergebnisspalten bestimmen
        For Each field As EasyResultField In hitTableHeader
            Dim sArray() As String
            sArray = header.Split(",")
            If sArray.Length > 0 Then
                For i As Integer = 0 To sArray.Length - 1
                    If field.Name.ToUpper = sArray(i).ToUpper Then
                        field.Index = CType(i + 1, String)
                        Exit For
                    End If
                Next
            End If
        Next

        With hitLst.Columns
            .Add("DOC_Location", System.Type.GetType("System.String"))
            .Add("DOC_Archive", System.Type.GetType("System.String"))
            .Add("DOC_ID", System.Type.GetType("System.String"))
            .Add("DOC_VERSION", System.Type.GetType("System.String"))
            .Add("Bilder", System.Type.GetType("System.String"))
            .Add("Link", System.Type.GetType("System.String"))
        End With

        For Each field As EasyResultField In hitTableHeader
            hitLst.Columns.Add(field.Name, GetType(String))
        Next

    End Sub

    Public Sub addHitTableRow(ByVal row As String, ByVal strLocFound As String, ByVal strArcFound As String, ByVal doc_id As String, ByVal doc_version As String)
        Dim nRow As Data.DataRow
        Dim index As Integer
        Dim value As String
        Dim found As Boolean
        Dim i As Integer

        row = row.Replace("^", "")
        row = row & ","
        nRow = hitLst.NewRow()
        nRow("DOC_Location") = strLocFound
        nRow("DOC_Archive") = strArcFound
        nRow("DOC_ID") = doc_id
        nRow("DOC_VERSION") = doc_version
        nRow("Bilder") = Nothing

        For Each field As EasyResultField In hitTableHeader
            index = 0
            found = False
            For i = 0 To row.Length - 1
                If row.Chars(i) = "," Then
                    index += 1
                End If
                If found = False Then
                    If (index = field.Index - 1) Then
                        If index = 0 Then
                            value = row.Substring(0, row.Length - i - 1)
                        Else
                            value = row.Substring(i + 1, row.Length - i - 1)
                        End If
                        '§§§ JVE 23.10.2006 TODO: Wenn nur 1 Spalte in der Trefferliste, nicht kommaseparierten String erwarten.
                        If value.Contains(",") Then
                            value = value.Substring(0, value.IndexOf(","))
                        End If

                        nRow(field.Name.ToUpper) = value
                        found = True
                    End If
                End If

            Next
        Next

        hitLst.Rows.Add(nRow)
        hitLst.AcceptChanges()
    End Sub
End Class

